task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -1147,8 +1151,22 @@ def final_state(self):
             return self._terminal_state or TaskInstanceState.SUCCESS
         if self._exit_code != 0 and self._terminal_state == SERVER_TERMINATED:
             return SERVER_TERMINATED
+
+        if self._exit_code != 0 and self._is_signal_retryable() and self._should_retry:
```

Comment:
```suggestion
        if and self._is_signal_retryable() and self._should_retry:
```
We already know the exit code is not 0 from the first condition of this fn on L1150.

Reply:
Makes sense.